src/main/java/lol/ui/Sprites.java

```diff
@@ -32,23 +58,33 @@ public ImageView grass() {
     return makeView(grassImage);
   }
 
-  public ImageView archer() {
-    return makeView(archerImage);
-  }
+  public ImageView championsView(String name, int teamID){
+    assert teamID < 2 : "Sprites are only provided for two teams at most.";
 
-  public ImageView warrior() {
-    return makeView(warriorImage);
+    if(name.equals("Archer")){
+      return makeView(archerImages[teamID]);
+    }
+    else if(name.equals("Warrior")){
+      return makeView(warriorImages[teamID]);
+    }
+    else {throw new RuntimeException("No champion named `" + name + "`.");}
   }
 
-  public ImageView blueNexus() {
-    return makeView(blueNexusImage);
+  public ImageView nexusesView(int teamID) {
```

Comment:
use an assert to keep it coherent with championsView (or use an exception in championView).
Or better, make a method `checkTeamID("Nexus", teamID)` that perform this check, this is to avoid code repetition.